gui/src/components/ProviderNetworks.tsx

```diff
@@ -0,0 +1,73 @@
+import { invoke } from "@tauri-apps/api/tauri";
+import { useRegisterActions } from "kbar";
+import { ReactNode, createContext } from "react";
+
+import { useInvoke } from "../hooks/tauri";
+import { useRefreshNetwork } from "../hooks/useRefreshNetwork";
+import { Network } from "../types";
+
+interface Value {
+  networks?: Network[];
+  network?: Network;
+  setNetwork: (newNetwork: string) => Promise<unknown>;
+  setNetworks: (newNetworks: Network[]) => Promise<unknown>;
+  resetNetworks: () => Promise<Network[]>;
+}
+
+export const NetworksContext = createContext<Value>({} as Value);
+
+const actionId = "network";
+
+export function ProviderNetworks({ children }: { children: ReactNode }) {
+  const { data: networks, mutate: mutateNetworks } =
+    useInvoke<Network[]>("networks_get_list");
+  const { data: network, mutate: mutateNetwork } = useInvoke<Network>(
+    "networks_get_current"
+  );
+
+  const value = {
+    networks,
+    network,
+    setNetwork: async (newNetwork: string) => {
+      if (network?.name === newNetwork) return;
+
+      await invoke("networks_set_current", { network: newNetwork });
+    },
+    setNetworks: async (newNetworks: Network[]) => {
+      await invoke("networks_set_list", { newNetworks: newNetworks });
+
+      mutateNetworks();
+    },
+    resetNetworks: async () => {
+      const networks: Network[] = await invoke("networks_reset");
+
+      mutateNetworks();
+
+      return networks;
+    },
+  };
```

Comment:
👍 honestly, the names are very bad :/ Having an "s" to distinguish things is no good. I'll take option 1 because option 2 is not solving the problem since destructing is a common pattern.